In [24]:
import re
import pyperclip

Creating a regex for phone numbers

In [25]:
PhoneRegex = re.compile(r'''
# Example phone numbers look like 415-555-1234, 555-1234 , (415) 555-1234 , 555-2222 ext 12345,ext. 12345, x12345
(((\d\d\d)|(\(\d\d\d\)))?  #area-code(optional)
(\s|-)           #first separator
\d\d\d            #first 3 digits
-                 #separator
\d\d\d\d           #last 4 digits
(((ext(\.)?\s)|x)      #Extension word-part (optional)
 (\d{2, 5}))?)
''', re.VERBOSE)

Creating a regex for email address

In [26]:
emailRegex = re.compile(r'''
#Example some+._thing@(\d{2,5}))?.com

[a-zA-Z0-9_.+]+  #name part - created own character class
@                # @ symbol
[a-zA-Z0-9_.+]+  # domain name part

''', re.VERBOSE)

Copy the whole document or selected paragraphs in to the clipboard and get it from pyperclip

In [27]:
text = pyperclip.paste()

In [28]:
print(text)

Directory of State Authorization Agencies and Lead Contacts
SHEEO – April 2014
U.S. STATES, District of Columbia, and Puerto Rico
ALABAMA
Alabama Commission on Higher Education
Elizabeth C. French
Director, Office of Institutional Effectiveness and Planning
Alabama Commission on Higher Education
Office of Institutional Effectiveness and
Planning 334-242-2179
elizabeth.french@ache.alabama.gov
Alabama Department of Postsecondary Education
Annette Funderburk
Private School Licensure Specialist
Alabama Department of Postsecondary
Education Office of Private School Licensing
Division 334-293-4651
annette.funderburk@dpe.edu
Alabama Office of the Secretary of State
Rebecca Morris, CPM
Director of Business Services
Office of Alabama Secretary of State
334-242-7221
rebecca.morris@sos.alabama.gov
ALASKA
Alaska Commission on Postsecondary Education
Kierke A. Kussart
Program Coordinator for Institutional Authorization
Alaska Commission on Postsecondary Education
907‐465‐6741
Kierke.kussart@alaska.

Extracting mail id and Phone number from text

In [29]:
extractedPhone = PhoneRegex.findall(text)
extractedEmail = emailRegex.findall(text)

print(extractedPhone)
print(extractedEmail)

[('334-242-2179', '334', '334', '', '-', '', '', '', '', ''), ('334-293-4651', '334', '334', '', '-', '', '', '', '', ''), ('334-242-7221', '334', '334', '', '-', '', '', '', '', ''), ('602-542-2399', '602', '602', '', '-', '', '', '', '', ''), ('501-371-2060', '501', '501', '', '-', '', '', '', '', ''), ('501-683-8000', '501', '501', '', '-', '', '', '', '', ''), ('916-431-6905', '916', '916', '', '-', '', '', '', '', ''), ('303-866-2723', '303', '303', '', '-', '', '', '', '', ''), ('303-866-2723', '303', '303', '', '-', '', '', '', '', ''), ('860-947-1822', '860', '860', '', '-', '', '', '', '', ''), ('302-857-3388', '302', '302', '', '-', '', '', '', '', ''), ('202-724-2095', '202', '202', '', '-', '', '', '', '', ''), ('850-245-3200', '850', '850', '', '-', '', '', '', '', ''), ('770-414-3208', '770', '770', '', '-', '', '', '', '', ''), ('208-332-1587', '208', '208', '', '-', '', '', '', '', ''), ('(217) 557-7352', '(217)', '', '(217)', ' ', '', '', '', '', ''), ('317-464-4400', 

In [30]:
allPhoneNumbers = []
for phoneNumber in extractedPhone:
    allPhoneNumbers.append(phoneNumber[0])
print(allPhoneNumbers)

['334-242-2179', '334-293-4651', '334-242-7221', '602-542-2399', '501-371-2060', '501-683-8000', '916-431-6905', '303-866-2723', '303-866-2723', '860-947-1822', '302-857-3388', '202-724-2095', '850-245-3200', '770-414-3208', '208-332-1587', '(217) 557-7352', '317-464-4400', '317-464-4400', '317-234-8338', '515-725-3413', '785-296-4917', '502-564-4185', '225-342-4253', '225-342-4253', '207-624-6607', '410-767-3268', '617-994-6922', '517-241-6806', '651-259-3975', '601-432-6372', '601-432-6185', '573-751-2361', '573-751-2361', '406-444-0607', '702-486-7330', '603-271-0257', '609-292-4310', '609-292-8052', '609-292-8658', '505-476-8418', '505-476-8402', '518-474-3969', '(919) 962-4550', '701-224-2498', '614-387-1215', '614-466-7802', '(405) 234-4300', '405-528-3370', '405-743-5198', '503-373-0003', '503-947-5716', '717-783-8228', '\n641-7100', '803-737-2281', '605-773-3537', '615-253-6285', '512-427-6527', '(866) 256-6333', '801-530-6601', '\n828-5402', '804-225-2629', '360-753-7869', '30

In [31]:
print(extractedEmail)

['elizabeth.french@ache.alabama.gov', 'annette.funderburk@dpe.edu', 'rebecca.morris@sos.alabama.gov', 'Kierke.kussart@alaska.gov', 'IA@alaska.gov', 'teri.stanfill@azppse.gov', 'Alana.Boles@adhe.edu', 'brenda.germann@arkansas.gov', 'joanne.wenzel@dca.ca.gov', 'heather.delange@dhe.state.co.us', 'Georgia.roberts@dhe.state.co.us', 'PSantoro@ctdhe.org', 'angeline.willen@doe.k12.de.us', 'robin.jenkins@dc.gov', 'sam.ferguson@fldoe.org', 'ccam@npec.state.ga.us', 'bobbi.lummew@dcca.hawaii.gov', 'valerie.fenske@osbe.idaho.gov', 'Cullen@ibhe.org', 'ksauer@che.in.gov', 'rmiller@bpe.che.in.gov', 'dedge1@dwd.in.gov', 'carolyn.small@iowa.gov', 'jjohnson@ksbor.org', 'sarah.levy@ky.gov', 'kcpe@ky.gov', 'carol.marabella@la.gov', 'larry.tremblay@la.gov', 'laurie.halligan@maine.gov', 'sackerball@mhec.state.md.us', 'approvalquery@bhe.mass.edu', 'beamishm@michigan.gov', 'george.roedler@state.mn.us', 'mdykes@mississippi.edu', 'kverneuille@mccb.edu', 'leroy.wade@dhe.mo.gov', 'rusty.monhollon@dhe.mo.gov', 'kja

Copying the extracted emails and phone numbers to the clipboard

In [32]:
result = '\n'.join(allPhoneNumbers) + '\n' + '\n'.join(extractedEmail)
pyperclip.copy(result)

copied output is 

334-242-2179
334-293-4651
334-242-7221
602-542-2399
501-371-2060
501-683-8000
916-431-6905
303-866-2723
303-866-2723
860-947-1822
302-857-3388
202-724-2095
850-245-3200
770-414-3208
208-332-1587
(217) 557-7352
317-464-4400
317-464-4400
317-234-8338
515-725-3413
785-296-4917
502-564-4185
225-342-4253
225-342-4253
207-624-6607
410-767-3268
617-994-6922
517-241-6806
651-259-3975
601-432-6372
601-432-6185
573-751-2361
573-751-2361
406-444-0607
702-486-7330
603-271-0257
609-292-4310
609-292-8052
609-292-8658
505-476-8418
505-476-8402
518-474-3969
(919) 962-4550
701-224-2498
614-387-1215
614-466-7802
(405) 234-4300
405-528-3370
405-743-5198
503-373-0003
503-947-5716
717-783-8228

641-7100
803-737-2281
605-773-3537
615-253-6285
512-427-6527
(866) 256-6333
801-530-6601

828-5402
804-225-2629
360-753-7869
304-558-0262
304-558-0262
304-558-2411
608-267-7733
307-777-6210
684-699-9155
680-488-1669
elizabeth.french@ache.alabama.gov
annette.funderburk@dpe.edu
rebecca.morris@sos.alabama.gov
Kierke.kussart@alaska.gov
IA@alaska.gov
teri.stanfill@azppse.gov
Alana.Boles@adhe.edu
brenda.germann@arkansas.gov
joanne.wenzel@dca.ca.gov
heather.delange@dhe.state.co.us
Georgia.roberts@dhe.state.co.us
PSantoro@ctdhe.org
angeline.willen@doe.k12.de.us
robin.jenkins@dc.gov
sam.ferguson@fldoe.org
ccam@npec.state.ga.us
bobbi.lummew@dcca.hawaii.gov
valerie.fenske@osbe.idaho.gov
Cullen@ibhe.org
ksauer@che.in.gov
rmiller@bpe.che.in.gov
dedge1@dwd.in.gov
carolyn.small@iowa.gov
jjohnson@ksbor.org
sarah.levy@ky.gov
kcpe@ky.gov
carol.marabella@la.gov
larry.tremblay@la.gov
laurie.halligan@maine.gov
sackerball@mhec.state.md.us
approvalquery@bhe.mass.edu
beamishm@michigan.gov
george.roedler@state.mn.us
mdykes@mississippi.edu
kverneuille@mccb.edu
leroy.wade@dhe.mo.gov
rusty.monhollon@dhe.mo.gov
kjanssen@montana.edu
kathleen.fimple@nebraska.gov
Brad.dirksen@nebraska.gov
dperlman@cpe.state.nv.us
patricia.edes@doe.nh.gov
Michelle.Robinson@njhe.state.nj.us
Robert.Grimmie@dol.state.nj.us
schoolapprovalunit@dol.state.nj.us
Diane.Vigil@state.nm.u
David.Mathews@state.nm.us
mborden@mail.nysed.gov
ezunon@mail.nysed.gov
studentcomplaint@northcarolina.edu
tanya.spilovoy@ndus.edu
sdegarmo@regents.state.oh.us
john.ware@scr.state.oh.us
gwekke@osrhe.edu
nhouse@obpvs.ok.gov
jrobi@okcareertech.org
alethia.miller@state.or.us
matt.barber@state.or.us
radistanceeducation@pa.gov
RACOLLUNIVSEMINFO@state.pa.us
izavala@ce.pr.gov
dvelletri@ribghe.org
reshleman@che.sc.gov
sos.edu@state.sd.us
lindsey.vaughan@tn.gov
teresa.warren@tn.gov
Cathie.Maeyaert@THECB.state.tx.us
catherine.bingle@twc.state.tx.us
career.schools@twc.state.tx.us
mwinegar@utah.gov
cathy.hilgendorf@state.vt.us
JoeDeFilippo@schev.edu
kareno@wsac.wa.gov
pspencer@wtb.wa.gov
stotler@hepc.wvnet.edu
stotler@hepc.wvnet.edu
tsmith@wvctcs.org
david.dies@eab.state.wi.us
elaine.marces@wyo.gov
k.kolhoff@amsamoa.edu
mben@comfsm.fm
mary.okada@guamcc.edu
anacletus2010@gmail.com
tellei@palau.edu
balikmad@yahoo.com
galving@nmcnet.edu
wwbenjamin@doe.vi